### PR TITLE
ci/release: Add release tooling

### PR DESCRIPTION
**Pull request by phlax**
This adds a bot workflow for creating release commits

As with similar tooling this is a first iteration to get things working - there is a lot that it doesnt do which are kinda necessary

This introduces some new release logic and and expectation that there is content in changelogs/summary.md 

The last author of that file will also be by default the signeroffer of the release

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .github/workflows/envoy-release.yml [hunk 1]
```diff
@@ -10,16 +10,132 @@ on:
...
+        - default
+        # only 1.25 does not have its own CI
+        # remove this and assoc logic when 1.25 goes EOL
+        - release/v1.25
+      pr:
+        type: boolean
+        default: true
+      pr_message:
+        description: Additional message for PR, eg to fix an issue (optional)
+      pr_signoff:
```
Comment (phlax) [inline .github/workflows/envoy-release.yml:37]: add author field separately

File: .github/workflows/envoy-release.yml [hunk 2]
```diff
@@ -10,16 +10,132 @@ on:
...
+  workflow_dispatch:
+    inputs:
+      task:
+        description: Select a task
+        required: true
+        default: create-release
+        type: choice
+        options:
+        - create-release
+        - sync-version-histories
```
Comment (phlax) [inline .github/workflows/envoy-release.yml:22]: remove

**Comment by phlax** [comment]
I tend to run this alot during release as changelogs change etc - so the first update to this wf will be to allow it to update as well as create 

I have also have a couple of prs behind for syncing version histories and reopening branches

File: .github/workflows/envoy-release.yml [hunk 3]
```diff
@@ -10,16 +10,133 @@ on:
...
+    inputs:
+      task:
+        description: Select a task
+        required: true
+        default: create-release
+        type: choice
+        options:
+          - create-release
+      # only 1.25 does not have its own CI
+      # remove this and assoc logic when 1.25 goes EOL
```
Comment (phlax) [inline .github/workflows/envoy-release.yml:23]: im thinking this is probs not necessary

**Review by RyanTheOptimist — APPROVED** [review]
Very exciting

**Pull request merged**